### PR TITLE
Differentiate noAccess form when unconnectable

### PR DESCRIPTION
**Pull request by Badlapje**
When an idp cannot request access a different form needs to be shown when clicking the no-access button.  This PR fixes that.

### CODE REVIEW

File: theme/base/javascripts/wayf/deleteDisable/setConnectability.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Ensures the right content/texts are shown for the noAccess form, depending on whether or not the clicked Idp is connectable or not.
+ *
+ * @param noAccess          the noAccess section element
+ * @param isConnectable     whether the clicked idp is connectable
+ */
+export const setConnectablity = (noAccess, isConnectable) => {
+  const isUnconnectable = noAccess.classList.contains('wayf__noAccess--unconnectable');
+
+  if (isUnconnectable && isConnectable) {
```
Comment (mroest) [inline theme/base/javascripts/wayf/deleteDisable/setConnectability.js:10]: 🙄 Maybe the names can be bit more specific/verbose. How can it be both unconnectable as connectable? :)

Comment (Badlapje) [reply theme/base/javascripts/wayf/deleteDisable/setConnectability.js]: done

File: theme/base/javascripts/wayf/handleClickingDisabledIdp.js [hunk 2]
```diff
@@ -4,14 +4,18 @@ import {setHiddenFieldValues} from './noAccess/setHiddenFieldValues';
...
+import {setConnectablity} from './deleteDisable/setConnectability';
+import {getData} from '../utility/getData';
 
 export const handleClickingDisabledIdp = (element) => {
   const noAccess = document.querySelector('.wayf__noAccess');
   const li = noAccess.querySelector('.wayf__idpList > li');
   const form = noAccess.querySelector('.noAccess__requestForm');
   const parentSection = element.closest('section');
   const cloneOfIdp = cloneIdp(element);
+  const isConnectable = getData(element, 'connectable') === 'true';
```
Comment (mroest) [inline theme/base/javascripts/wayf/handleClickingDisabledIdp.js:16]: i would suggest to name this `connectable`

File: theme/base/javascripts/wayf/deleteDisable/setConnectability.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Ensures the right content/texts are shown for the noAccess form, depending on whether or not the clicked Idp is connectable or not.
+ *
+ * @param noAccess          the noAccess section element
+ * @param isConnectable     whether the clicked idp is connectable
+ */
+export const setConnectablity = (noAccess, isConnectable) => {
```
Comment (mroest) [inline theme/base/javascripts/wayf/deleteDisable/setConnectability.js]: i would rename to `connectable`

File: theme/base/javascripts/wayf/deleteDisable/setConnectability.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  const isUnconnectable = noAccess.classList.contains('wayf__noAccess--unconnectable');
```
Comment (mroest) [inline theme/base/javascripts/wayf/deleteDisable/setConnectability.js]: i would use something like `currentState`

File: theme/base/javascripts/wayf/deleteDisable/setConnectability.js (hunk 1 above)
Comment (mroest) [reply theme/base/javascripts/wayf/deleteDisable/setConnectability.js]: then this would be `if currentstate != connectable`

**Review by mroest — APPROVED** [review]

File: theme/base/javascripts/wayf/deleteDisable/setConnectability.js (hunk 1 above)
Comment (Badlapje) [reply theme/base/javascripts/wayf/deleteDisable/setConnectability.js]: this is sheer beauty 😄  🥇

**Pull request merged**